OpenRA.Game/Network/TickTime.cs

```diff
@@ -0,0 +1,52 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+
+namespace OpenRA.Network
+{
+	public sealed class TickTime
+	{
+		readonly Func<int> timestep;
+
+		long lastTickTime;
+
+		public TickTime(Func<int> timestep, long lastTickTime)
+		{
+			this.timestep = timestep;
+			this.lastTickTime = lastTickTime;
+		}
+
+		public long Value
+		{
+			get => lastTickTime;
+			set => lastTickTime = value;
+		}
+
+		public bool ShouldUpdate(long tick)
```

Comment:
Style nit: `ShouldUpdate` is a bit odd when we don't have an `Update` method to call. How about `ShouldAdvance`?